### PR TITLE
Add type assertion for `.extend()` argument

### LINKED ISSUE #40: Maybe some assertions for `defaults` type in `extend`?
The `defaults` param will be used to `deepMerge` with other options and the `deepMerge` method will flat items for array argument.
Everything goes on normal and users may not know what happen if they pass in an array until they look into code. But there is something changed for options for `Ky` constructor actually.
Maybe there could be some assertions for the `defaults` param type as it should be an object? :)

### PR DESCRIPTION
**Pull request by poppinlp**
Fix #40

### CODE REVIEW

File: test/main.js [hunk 1]
```diff
@@ -179,6 +179,22 @@ test('ky.extend()', async t => {
...
+		666,
+		'hello',
+		[],
+		null,
+		() => {},
+		Symbol('ky')
+	].forEach(value => {
+		t.throws(() => {
+			ky.extend(value);
+		});
```
Comment (sindresorhus) [inline test/main.js]: You should also assert that it throws the correct message and the type of the error. Use the object form of the `t.throws` assertions: https://github.com/avajs/ava#throwsasyncthrower-expected-message

File: test/main.js [hunk 2]
```diff
@@ -179,6 +179,22 @@ test('ky.extend()', async t => {
...
+test('ky.extend() throws when given non-object argument', t => {
+	[
+		true,
+		666,
+		'hello',
+		[],
+		null,
+		() => {},
+		Symbol('ky')
+	].forEach(value => {
```
Comment (sindresorhus) [inline test/main.js]: Make the list of types a separate variable and use a `for-of` loop instead. (Just my preference. Makes it more readable)

**Comment by poppinlp** [comment]
@sindresorhus Hi. I've pushed a commit for that. Thanks for your aborative review.

**Comment by sindresorhus** [comment]
Like I commented in https://github.com/sindresorhus/ky/pull/50#discussion_r223930968, you can use the object form of the throws assertion so you don't have to use `t.is` to test the error message.

File: test/main.js [hunk 3]
```diff
@@ -179,6 +179,24 @@ test('ky.extend()', async t => {
...
+		[],
+		null,
+		() => {},
+		Symbol('ky')
+	];
+
+	for (const value of nonObjectValues) {
+		t.throws(() => {
+			ky.extend(value);
+		}, TypeError, 'The `defaultOptions` argument must be an object');
```
Comment (sindresorhus) [inline test/main.js]: Still not the **object** form. Should be:

```js
t.throws(
	() => {
		ky.extend(value);
	}
	{
		instanceOf: TypeError,
		message: 'The `defaultOptions` argument must be an object'
	}
);
```

**Comment by poppinlp** [comment]
Sorry that's my fault. I've seen the `.throws(fn, [expected, [message]])` and have not looked into the description of `expected` argument.

**Comment by poppinlp** [comment]
By the way, maybe there could be an error helper which includes our all static errors. We could require it in code and using to throw errors in lib and do assertions in test cases. This will prevent the same error message appearing in two places which we must keep them same right now. And this may be helpful for i18n(not the key point).

**Comment by sindresorhus** [comment]
Could make sense in the future if we end up with a lot of error messages, but for now, it feels like a lot of overhead for little gain. The messages won't change much anyway, so I think it's fine how it is.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**